src/pages/company-info/index.js

```diff
@@ -0,0 +1,211 @@
+import { Box, Grid } from '@mui/material'
+import { useFormik } from 'formik'
+import React, { useContext, useEffect, useState } from 'react'
+import CustomImage from 'src/components/Inputs/CustomImage'
+import CustomTextField from 'src/components/Inputs/CustomTextField'
+import FormShell from 'src/components/Shared/FormShell'
+import ResourceComboBox from 'src/components/Shared/ResourceComboBox'
+import useResourceParams from 'src/hooks/useResourceParams'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { SystemRepository } from 'src/repositories/SystemRepository'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import toast from 'react-hot-toast'
+import { useForm } from 'src/hooks/form'
+
+const CompanyInfo = () => {
+  const [initialValues, setInitialData] = useState({
+    plantId: '',
+    accountId: JSON.parse(window.sessionStorage.getItem('userData')).accountId,
+    name: '',
+    webSite: '',
+    taxNo: '',
+    licenseNo: '',
+    crNo: '',
+    logoUrl: '',
+    flName: '',
+    attachment: null
+  })
+  const { getRequest, postRequest } = useContext(RequestsContext)
+
+  const { labels: labels, access: maxAccess } = useResourceParams({
+    datasetId: ResourceIds.CompanyInfo
+  })
+
+  // const [file, setFile] = useState()
+
+  useEffect(() => {
+    getData()
+  }, [])
+
+  async function getData() {
+    const res = await getRequest({
+      extension: SystemRepository.CompanyInfo.get,
+      parameters: `_filter=`
+    })
+
+    const result = await getRequest({
+      extension: SystemRepository.Attachment.get,
+      parameters: `_resourceId=20120&_seqNo=0&_recordId=1`
+    })
+    res.record.attachment = result.record
+    res.record.accountId = JSON.parse(window.sessionStorage.getItem('userData')).accountId
+    setInitialData(res.record)
+  }
+
+  const { formik } = useForm({
+    maxAccess,
+    enableReinitialize: true,
+    validateOnChange: true,
+    initialValues,
+    onSubmit: values => {
+      post(values)
+    }
+  })
+
+  const post = obj => {
+    postRequest({
+      extension: SystemRepository.CompanyInfo.set,
+      record: JSON.stringify({ ...obj, logoUrl: null })
+    })
+      .then(res => {
+        if (res && !file) toast.success('Record Edited Successfully')
+      })
+      .catch(error => {})
+
+    if (obj?.logoUrl) {
+      postRequest({
+        extension: SystemRepository.Attachment.set,
+        record: JSON.stringify(obj.attachment),
+        file: obj.logoUrl
+      }).then(res => {
+        if (res) toast.success('Record Edited Successfully')
+      })
+    } else {
+      initialValues.logoUrl &&
+        postRequest({
```

Comment:
this request should be called from the new CustomImage component. you send as props the resourceId, recordId, seqNo.

Reply:
Yes.